nilearn/plotting/surf_plotting.py

```diff
@@ -3,18 +3,186 @@
 Only matplotlib is required.
 """
 
-# Import libraries
 import nibabel
 import numpy as np
+from scipy import sparse, interpolate
+
+# These will be removed (see comment on _points_in_unit_ball)
+from sklearn.externals import joblib
+from ..datasets.utils import _get_dataset_dir
+import sklearn.cluster
+import sklearn.preprocessing
+# / These will be removed (see comment on _points_in_unit_ball)
+
 import matplotlib.pyplot as plt
 
 from mpl_toolkits.mplot3d import Axes3D
 from nibabel import gifti
 
+import nilearn.image
 from .._utils.compat import _basestring
+from .. import _utils
 from .img_plotting import _get_colorbar_and_data_ranges
 
 
+# Eventually, when this PR is ready, the sample locations inside the unit ball
+# will be hardcoded. For now we just compute them in a simple way (we will find
+# something better than the k-means hack) and cache the result.
+memory = joblib.Memory(_get_dataset_dir('joblib'), verbose=False)
+
+
+@memory.cache
+def _uniform_ball_cloud(n_points=100, dim=3, n_monte_carlo=5000):
+    mc_cube = np.random.uniform(-1, 1, size=(n_monte_carlo, dim))
+    mc_ball = mc_cube[(mc_cube**2).sum(axis=1) <= 1.]
+    centroids, assignments, _ = sklearn.cluster.k_means(
+        mc_ball, n_clusters=n_points)
+    return centroids
+
+
+# could be replaced by nilearn.image.resampling.coord_transform, but
+# it only works for 3d images and 2d is useful for visu and debugging.
+def _transform_coord(coords, affine):
+    return affine.dot(np.vstack([coords.T, np.ones(coords.shape[0])]))[:-1].T
+
+
+def _face_normals(mesh):
+    vertices, faces = load_surf_mesh(mesh)
```

Comment:
Usually when faces in a triangular mesh are described by their ordered vertices,
the normal given by the right-hand rule points "outwards". I think that this
convention is respected by cortical meshes because that is what matplotlib
expects, and surf_plotting.plot_surf passes the faces to matplotlib without
modifying them (and it works fine). Can anyone confirm? I will make this explicit.